CMakeLists.txt

```diff
@@ -0,0 +1,299 @@
+        else()
+            target_compile_options(${_target} PRIVATE ${CMOCKA_STATIC_CFLAGS})
+            target_link_libraries(${_target} PRIVATE ${CMOCKA_STATIC_LIBRARIES})
+        endif()
+    endforeach()
+
+    target_link_options(test_lurch_api PRIVATE
+        -Wl,--wrap=purple_account_get_username
+        -Wl,--wrap=purple_account_get_connection
+        -Wl,--wrap=purple_connection_get_protocol_data
+        -Wl,--wrap=purple_signal_register
+        -Wl,--wrap=purple_signal_unregister
+        -Wl,--wrap=purple_signal_connect
+        -Wl,--wrap=purple_signal_disconnect
+        -Wl,--wrap=purple_find_conversation_with_account
+        -Wl,--wrap=jabber_pep_publish
+        -Wl,--wrap=jabber_chat_find_by_conv
+        -Wl,--wrap=jabber_iq_send
+        -Wl,--wrap=axc_get_device_id
+        -Wl,--wrap=axc_key_load_public_own
+        -Wl,--wrap=axc_key_load_public_addr
+        -Wl,--wrap=axc_session_exists_any
+        -Wl,--wrap=omemo_storage_user_devicelist_retrieve
+        -Wl,--wrap=omemo_storage_chatlist_delete
+        -Wl,--wrap=omemo_storage_chatlist_save
+        -Wl,--wrap=omemo_storage_chatlist_exists
+        -Wl,--wrap=omemo_storage_user_devicelist_retrieve
+        -Wl,--wrap=lurch_util_fp_get_printable
+    )
+
+    target_link_options(test_lurch_crypto PRIVATE
+        -Wl,--wrap=axc_message_encrypt_and_serialize
+        -Wl,--wrap=axc_session_exists_initiated
+    )
+
+    target_link_options(test_lurch_util PRIVATE
+        -Wl,--wrap=purple_user_dir
+        -Wl,--wrap=purple_prefs_get_bool
+        -Wl,--wrap=purple_prefs_get_int
+        -Wl,--wrap=purple_debug_error
+        -Wl,--wrap=purple_debug_info
+        -Wl,--wrap=purple_debug_misc
+        -Wl,--wrap=purple_base16_encode_chunked
+    )
+endif()
+
+
+#
+# External build dependencies
+#
+foreach(_target lurch lurch_internal ${_LURCH_TEST_TARGETS})
+    target_include_directories(${_target} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/headers/jabber)
+
+    if(LURCH_WITH_SYSTEM_AXC)
+        if(BUILD_SHARED_LIBS)
+            target_compile_options(${_target} PRIVATE ${AXC_CFLAGS})
+            target_link_libraries(${_target} PRIVATE ${AXC_LIBRARIES})
+        else()
+            target_compile_options(${_target} PRIVATE ${AXC_STATIC_CFLAGS})
+            target_link_libraries(${_target} PRIVATE ${AXC_STATIC_LIBRARIES})
+        endif()
+    else()
+        target_link_libraries(${_target} PRIVATE axc)
+    endif()
+
+    if(LURCH_WITH_SYSTEM_OMEMO)
+        if(BUILD_SHARED_LIBS)
+            target_compile_options(${_target} PRIVATE ${OMEMO_CFLAGS})
+            target_link_libraries(${_target} PRIVATE ${OMEMO_LIBRARIES})
+        else()
+            target_compile_options(${_target} PRIVATE ${OMEMO_STATIC_CFLAGS})
+            target_link_libraries(${_target} PRIVATE ${OMEMO_STATIC_LIBRARIES})
+        endif()
+    else()
+        target_link_libraries(${_target} PRIVATE omemo)
+    endif()
+
+    if(BUILD_SHARED_LIBS)
```

Comment:
is there a way to more clearly name the output dll? right now, after unpacking the result of the zip files, it is not clear whether the `lurch.dll`already contains the submodules or not.

Reply:
Do we really want to change the name based on that? Can I vote against it? :smiley: It would not be hard, but it "feels wrong".

Reply:
@gkdr while I'm still unsure if that's a good move, I have make the build system create either `lurch_fully_shared.dll` or `lurch_semi_shared.dll` now, to address your wish for distinguishability.

Reply:
why does it feel wrong to you? do you have a different suggestion for making this more clear?

Reply:
It's the mix of the ugly names, the fact that it's not done at most places, and that there are other means to find out or be sure which file you have, e.g. running `objdump`. It just feels like solving a problem that can be avoided on other layers and once the plugin is running, it won't matter much which one it is. But that's my personal view.

Reply:
thanks for implementing it :slightly_smiling_face: